### PR TITLE
Db/gen remaining structs

### PR DESCRIPTION
**Pull request by davidbloss**
## Issues

[#229](https://github.com/OpsLevel/team-platform/issues/229)

## Changelog

Update `gen.go` to generate almost everything. This PR focuses on the generated `interfaces.go` file.
This file contains all structs that map to GraphQL objects of type interface.

`CustomActionsExternalAction`, `Check`, and `Integration` have been replaced by generated code.

### Secondary Changes (future PRs)

Overview for future PRs:
- `object.go` contains the vast majority of data structures
- `connection.go` contains all `*Connection` and `*Edge` structs
- `payload.go` contains all the returned payload types from GraphQL mutations.
- `query.go` contains all the `Get` and `List` resource operations
- `mutation.go` contains all the `Create`, `Update`, `Delete` resource operations

- [X] List your changes here
- [X] Make a `changie` entry

## Tophatting

`task gen` creates `interfaces.go` (and others, ignored here)

### CODE REVIEW

File: check_test.go [hunk 1]
```diff
@@ -37,7 +37,7 @@ func getCheckTestCases() map[string]TmpCheckTestCase {
...
...ame,predicates{key,keyData,type,value,caseSensitive}},id,level{alias,description,id,index,name},name,notes: rawNotes,owner{... on Team{alias,id}},type,... on AlertSourceUsageCheck{alertSourceNamePredicate{type,value},alertSourceType},... on CustomEventCheck{integration{id,name,type},passPending,resultMessage,serviceSelector,successCondition},... on HasRecentDeployCheck{days},... on ManualCheck{updateFrequency{startingDate,frequencyTimeScale,frequencyValue},updateRequiresComment},... on RepositoryFileCheck{directorySearch,filePaths,fileContentsPredicate{type,value},useAbsoluteRoot},... on RepositoryGrepCheck{directorySearch,filePaths,fileContentsPredicate{type,value}},... on RepositorySearchCheck{fileExtensions,fileContentsPredicate{type,value}},... on ServiceOwnershipCheck{requireContactMethod,contactMethod,tagKey,tagPredicate{type,value}},... on ServicePropertyCheck{serviceProperty,propertyValuePredicate{type,value}},... on TagDefinedCheck{tagKey,tagPredicate{type,value}},... on ToolUsageCheck{toolCategory,toolNamePredicate{type,value},toolUrlPredicate{type,value},environmentPredicate{type,value}},... on HasDocumentationCheck{documentType,documentSubtype}},errors{message,path}}}`,
```
Comment (davidbloss) [inline check_test.go:40]: All changes in this file come from the ordering in the `Check` struct.
The fields `Enabled` and `EnableOn` swapped places in the struct, so the queries here have the same swap to match.

Was: `enabled,enableOn`
Is now: `enableOn,enabled`

File: actions.go [hunk 2]
```diff
@@ -9,16 +9,6 @@ type CustomActionsId struct {
 	Id      ID       `graphql:"id"`
 }
```
Comment (davidbloss) [inline actions.go:11]: Moved to `interfaces.go`

**Review by taimoorgit — COMMENTED** [review]
Looks good so far. My preliminary comments:

1. Templates take a large part of gen.go and it would be better to move them to a separate file. 
1. There's some commented out code gen.go code that should be deleted altogether that was not added in this PR. For example see the mentions of `scalarFile` and `unionFile` which are never used.

File: gen.go [hunk 3]
```diff
@@ -22,56 +23,105 @@ import (
...
-	// payloadFile     string = "pkg/gen/payload.go"
-	// queryFile       string = "pkg/gen/query.go"
-	// scalarFile      string = "pkg/gen/scalar.go"
-	// unionFile       string = "pkg/gen/union.go"
+	interfacesFile  string = "interfaces.go"
+	objectFile      string = "object.go"
+	queryFile       string = "query.go"
+	mutationFile    string = "mutation.go"
+	payloadFile     string = "payload.go"
+	// scalarFile      string = "scalar.go" // NOTE: probably not useful
```
Comment (taimoorgit) [inline gen.go:34]: Would prefer if we just deleted these now

File: gen.go [hunk 4]
```diff
@@ -201,22 +251,22 @@ func run() error {
...
+			subSchema = interfaceSchema
+		case objectFile:
+			subSchema = objectSchema
+		case mutationFile:
+			subSchema = objectSchema
+		case payloadFile:
+			subSchema = objectSchema
+		case queryFile:
+			subSchema = objectSchema
 		// case scalarFile:
```
Comment (taimoorgit) [inline gen.go:270]: Same here

**Review by taimoorgit — APPROVED** [review]
+1 for merging this now

File: gen.go (hunk 3 above)
Comment (davidbloss) [reply gen.go:34]: There is a good chunk of work left in `gen.go`. I'm not 100% sure that these parts are useless so I'll wait to remove for now

Comment (taimoorgit) [reply gen.go:34]: got it

**Pull request merged**